xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1056 @@
+from __future__ import annotations
```

Comment:
would be great to have your PR description as a docstring at the top of this file, I'm sure future readers will find it useful, especially external links